server/src/main/java/org/opensearch/env/NodeRepurposeCommand.java

```diff
@@ -94,85 +98,186 @@ protected boolean validateBeforeLock(Terminal terminal, Environment env) {
+            Sets.union(
+                indexUUIDsFor(indexPaths),
+                StreamSupport.stream(metadata.indices().values().spliterator(), false)
+                    .map(imd -> imd.value.getIndexUUID())
+                    .collect(Collectors.toSet())
+            )
         );
 
-        outputVerboseInformation(terminal, indexPaths, indexUUIDs, metadata);
-
-        terminal.println(noClusterManagerMessage(indexUUIDs.size(), shardDataPaths.size(), indexMetadataPaths.size()));
+        List<Path> cleanUpPaths = new ArrayList<>(shardDataPaths);
+        cleanUpPaths.addAll(fileCacheDataPaths);
+        outputVerboseInformation(terminal, cleanUpPaths, indexUUIDs, metadata);
+        terminal.println(noClusterManagerMessage(indexUUIDs.size(), cleanUpPaths.size(), indexMetadataPaths.size()));
         outputHowToSeeVerboseInformation(terminal);
 
-        terminal.println("Node is being re-purposed as no-cluster-manager and no-data. Clean-up of index data will be performed.");
+        if (repurposeData && repurposeSearch) {
+            terminal.println(
+                "Node is being re-purposed as no-cluster-manager, no-data and no-search. Clean-up of index data and file cache will be performed."
+            );
+        } else if (repurposeData) {
+            terminal.println("Node is being re-purposed as no-cluster-manager and no-data. Clean-up of index data will be performed.");
+        } else if (repurposeSearch) {
+            terminal.println(
+                "Node is being re-purposed as no-cluster-manager and no-search. Clean-up of file cache and corresponding index metadata will be performed."
+            );
+        }
         confirm(terminal, "Do you want to proceed?");
 
-        removePaths(terminal, indexPaths); // clean-up shard dirs
         // clean-up all metadata dirs
         MetadataStateFormat.deleteMetaState(dataPaths);
-        IOUtils.rm(Stream.of(dataPaths).map(path -> path.resolve(INDICES_FOLDER)).toArray(Path[]::new));
+        if (repurposeData) {
+            removePaths(terminal, indexPaths); // clean-up shard dirs
+            IOUtils.rm(Stream.of(dataPaths).map(path -> path.resolve(INDICES_FOLDER)).toArray(Path[]::new));
+        }
+
+        if (repurposeSearch) {
+            removePaths(terminal, fileCachePaths); // clean-up file cache dirs
+            IOUtils.rm(dataPaths[FILE_CACHE_NODE_PATH_LOCATION].resolve(CACHE_FOLDER));
+        }
 
-        terminal.println("Node successfully repurposed to no-cluster-manager and no-data.");
+        if (repurposeData && repurposeSearch) {
+            terminal.println("Node successfully repurposed to no-cluster-manager, no-data and no-search.");
+        } else if (repurposeData) {
+            terminal.println("Node successfully repurposed to no-cluster-manager and no-data.");
+        } else if (repurposeSearch) {
+            terminal.println("Node successfully repurposed to no-cluster-manager and no-search.");
+        }
```

Comment:
Seems like you could avoid this explosion of cases with something like:

```final List<String> roleText = new ArrayList<>();
roleText.add("no-cluster-manager");
if (repurposeData) roleText.add("no-data");
if (repurposeSearch) roleText.add("no-search");
terminal.println("Node successfully repurposed to: " + String.join(",", roleText));
```

Same applies in the other cases though might require more refactoring. What do you think?

Reply:
I thought of doing that, but this makes it more readable and traceable. Given its use case and frequency of use, I think we would rather benefit from this approach.